src/box/memtx_space.c

```diff
@@ -869,11 +869,12 @@ memtx_space_check_index_def(struct space *space, struct index_def *index_def)
 	}
 
 	/* Only HASH and TREE indexes checks parts there */
-	/* Check that there are no ANY, ARRAY, MAP parts */
 	for (uint32_t i = 0; i < key_def->part_count; i++) {
 		struct key_part *part = &key_def->parts[i];
-		if (part->type <= FIELD_TYPE_ANY ||
-		    part->type >= FIELD_TYPE_INTERVAL) {
+		if (part->type == FIELD_TYPE_ANY ||
+		    part->type == FIELD_TYPE_INTERVAL ||
+		    part->type == FIELD_TYPE_ARRAY ||
+		    part->type == FIELD_TYPE_MAP) {
```

Comment:
Better add a helper function for this check.

Reply:
Done.